### PR TITLE
v0.3, switched to XML for parsing

### PR DESCRIPTION
**Pull request by apBUSampK**
Mostly the changes are separation of declaration and definition and inclusion of [tinyxml2](https://github.com/leethomason/tinyxml2) for better parsing. Also some changes to API to improve it (string to const string&, maps for Factories instead of plain strings, etc.)

### CODE REVIEW

**Comment by apBUSampK** [comment]
Also C++17 cause looks like we can switch to that on software we use.

File: COLA-0.3/tinyxml2/test/CMakeLists.txt
Comment (alexsvetlichnyy) [inline COLA-0.3/tinyxml2/test/CMakeLists.txt:1]: Why empty?

File: COLA.cc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            case FilterType::writer:
+                regWrite(std::move(factory), name);
+                break;
+            default:
+                throw std::domain_error("ERROR in MetaProcessor: No such type of filter.");
+        }
+    }
+
+    std::map<std::string, std::string> _get_name_and_params (const tinyxml2::XMLElement* element, std::string& name) {
+        auto currentAttribute = element->FindAttribute("name");
```
Comment (alexsvetlichnyy) [inline COLA.cc:67]: trying to find string "name"

File: COLA.cc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                ensemble.converters.push_back(std::unique_ptr<VConverter>(dynamic_cast<VConverter*>(converterMap.at(name)->create(params))));
+                params.clear();
+                currentElement = currentElement->NextSiblingElement();
+            }
+
+            params = _get_name_and_params(currentElement, name);
+            ensemble.writer = std::unique_ptr<VWriter>(dynamic_cast<VWriter*>(writerMap.at(name)->create(params)));
+            return ensemble;
+        } else {
+            throw std::runtime_error("ERROR in MetaProcessor: Couldn't open XML file " + fname + "\nError code: " +
```
Comment (alexsvetlichnyy) [inline COLA.cc]: better add "is it XML file?"

**Review by alexsvetlichnyy — CHANGES_REQUESTED** [review]

File: COLA.cc (hunk 1 above)
Comment (apBUSampK) [reply COLA.cc:67]: As intended, description starts with name, then all other attributes. Name is mandatory.

File: COLA.cc (hunk 2 above)
Comment (apBUSampK) [reply COLA.cc]: Changed now. The error can happen due to other reasons, code should tell it all.

File: COLA-0.3/tinyxml2/test/CMakeLists.txt
Comment (apBUSampK) [reply COLA-0.3/tinyxml2/test/CMakeLists.txt:1]: Shouldn't even be there, removed

File: include/COLA.hh
Comment (alexsvetlichnyy) [inline include/COLA.hh:1]: So, you left include dir empty?

Comment (apBUSampK) [reply include/COLA.hh:1]: Yes, COLA.hh is now in root directory of project.

If this is somehow not the case I might have broken git on my pc.

**Review by alexsvetlichnyy — APPROVED** [review]

**Pull request merged**